src/io.h

```diff
@@ -0,0 +1,484 @@
+**  'file' holds the file identifier which is  received  from  'SyFopen'  and
+**  which is passed to  'SyFputs'  and  'SyFclose'  to  identify  this  file.
+**  'line' is a buffer that holds the current output line.
+**  'pos' is the position of the current character on that line.
+**
+**  'OutputFiles' is the stack of open output files.  It  is  represented  as
+**  an array of structures of type 'TypOutputFile'.
+**
+**  'Output' is a pointer to the current output file.  It points to  the  top
+**  of the stack 'OutputFiles'.
+*/
+/* the widest allowed screen width */
+#define MAXLENOUTPUTLINE 4096
+/* the maximal number of used line break hints */
+#define MAXHINTS 100
+typedef struct {
+    UInt isstream;
+    UInt isstringstream;
+    Int  file;
+    Char line[MAXLENOUTPUTLINE];
+    Int  pos;
+    Int  format;
+    Int  indent;
+    /* each hint is a tripel (position, value, indent) */
+    Int hints[3 * MAXHINTS + 1];
+    Obj stream;
+} TypOutputFile;
+
+/****************************************************************************
+**
+*F  GetCurrentOutput()  . . . . . . . . . . . get the current thread's output
+**
+**  The same as 'OpenOutput' but for streams.
+*/
+extern TypOutputFile * GetCurrentOutput(void);
+
+/****************************************************************************
+**
+*F  Pr( <format>, <arg1>, <arg2> )  . . . . . . . . .  print formatted output
+**
+**  'Pr' is the output function. The first argument is a 'printf' like format
+**  string containing   up   to 2  '%'  format   fields,   specifing  how the
+**  corresponding arguments are to be  printed.  The two arguments are passed
+**  as  'long'  integers.   This  is possible  since every  C object  ('int',
+**  'char', pointers) except 'float' or 'double', which are not used  in GAP,
+**  can be converted to a 'long' without loss of information.
+**
+**  The function 'Pr' currently support the following '%' format  fields:
+**  '%c'    the corresponding argument represents a character,  usually it is
+**          its ASCII or EBCDIC code, and this character is printed.
+**  '%s'    the corresponding argument is the address of  a  null  terminated
+**          character string which is printed.
+**  '%d'    the corresponding argument is a signed integer, which is printed.
+**          Between the '%' and the 'd' an integer might be used  to  specify
+**          the width of a field in which the integer is right justified.  If
+**          the first character is '0' 'Pr' pads with '0' instead of <space>.
+**  '%>'    increment the indentation level.
+**  '%<'    decrement the indentation level.
+**  '%%'    can be used to print a single '%' character. No argument is used.
+**
+**  You must always  cast the arguments to  '(long)' to avoid  problems  with
+**  those compilers with a default integer size of 16 instead of 32 bit.  You
+**  must pass 0L if you don't make use of an argument to please lint.
+*/
+
+typedef TypOutputFile * KOutputStream;
```

Comment:
Future note: get rid of this typedef -- reading `TypOutputFile *` is easier to understand for me than reading `KOutputStream`.

We then also might want to rename `TypOutputFile` to `TypOutputStream` or just `OutputStream` (and similar for input).